lib/datadog/kit.rb

```diff
@@ -0,0 +1,9 @@
+# typed: false
+
+module Datadog
+  # Namespace for Datadog Kit
+  module Kit
```

Comment:
I'm curious, where does this "Kit" namespace comes from, and how does it differ from "Core" or the others already in? (At least for me it isn't entirely self-explanatory.)

Reply:
To develop on this further, given the quite innocuous changeset regarding `set_user`, I think the introduction of `Datadog::Kit` is the true goal of this PR: is that sensible, both for us structurally and as a public API?

Reply:
Thanks for the explanation.

Naming is hard 😄 ! I tried to come up with some ideas but it's a bit hard to generalize from something that only contains a single API. I guess let's see where this leads us for now 😉 

(I'm still unconvinced that this `Identity` in particular this shouldn't be inside Tracing, but I don't mind to disagree and commit on it, since you and David are happy with this path)

@lloeki would you mind attaching your explanation to the `kit.rb` file? That way newer contributors and colleagues can also understand why we created it in the future.

Reply:
Sure